File: src/main/java/uk/yermak/audiobookconverter/ProgressCallback.java
package uk.yermak.audiobookconverter;

import uk.yermak.audiobookconverter.fx.ConversionProgress;

public class ProgressCallback {
    private final String fileName;
    private final ConversionProgress conversionProgress;

    public String fileName() {
        return this.fileName;
    }

    public void converted(final long timeInMillis, final long size) {
        this.conversionProgress.converted(this.fileName(), timeInMillis, size);
    }

    public void completedConversion() {
        this.conversionProgress.incCompleted(this.fileName());
    }

    public void reset() {
        this.conversionProgress.reset();
    }

    public ProgressCallback(final String fileName, final ConversionProgress refreshable) {
        this.fileName = fileName;
        this.conversionProgress = refreshable;
    }

    public void setState(String message) {
        conversionProgress.setState(message);
    }
}

        

File: src/main/scala/uk/yermak/audiobookconverter/ProgressCallback.scala
package uk.yermak.audiobookconverter

/**
  * Created by Yermak on 03-Jan-18.
  */
class ProgressCallback(var fileName: String, var refreshable: Refreshable) {
  def converted(timeInMillis: Long, size: Long): Unit = refreshable.converted(fileName, timeInMillis, size)

  def completedConversion(): Unit = refreshable.incCompleted(fileName)

  def reset(): Unit = refreshable.reset()
}